doc/DHCPv6_Relay/DHCPv6_Relay_HLD.md

```diff
@@ -0,0 +1,155 @@
+  - [2.2 DHCP Relay for IPv6 process in dhcp-relay container](#22-dhcp-relay-for-ipv6-process-in-dhcp-relay-container)
+  - [2.3 DHCP Monitor](#23-dhcp-monitor)
+- [3 CLI](#3-cli)
+- [4 Init flow](#4-init-flow)
+
+# List of Tables
+* [Table 1: Abbreviations](#definitionsabbreviation)
+
+# List of Figures
+* [DHCPv6 Diagram](#2-modules-design)
+* [DHCPv6 init flow](#4-init-flow)
+
+# Revision
+| Rev | Date     | Author          | Change Description                 |
+|:---:|:--------:|:---------------:|------------------------------------|
+| 0.1 | 03/04    | Shlomi Bitton   | Initial version                    |
+
+# About this Manual
+This document provides an overview of the implementation and integration of DHCP Relay for IPv6 feature in SONiC.
+
+# Scope
+This document describes the high level design of the DHCP Relay for IPv6 feature in SONiC.
+
+# Definitions/Abbreviation
+| Abbreviation  | Description                               |
+|---------------|-------------------------------------------|
+| DHCP          | Dynamic Host Configuration Protocol       |
+
+# 1 Requirements Overview
+
+## 1.1 Functional Requirements
+
+DHCP Relay for IPv6 feature in SONiC should meet the following high-level functional requirements:
+
+- Give the support for relaying DHCP packets from downstream networks to upstream networks using IPv6 addresses.
+- Provide the functionality as a seperate process running on dhcp-relay docker container.
+- Relaying messages to multiple unicast and multicast addresses.
+
+## 1.2 Configuration and Management Requirements
+
+- DHCPv6 trap should be enabled through the COPP manager when the DHCP relay feature is enabled and vice versa. 
+- Downstream network is the VLAN interface with the relay configuration. Global IPv6 address is required to be configured on that interface. 
+- Config DB schema should meet the following format:
+```
+{
+"VLAN": {
+  "Vlan1000": {
+    "dhcp_servers": [
+      "192.0.0.1", 
+      "192.0.0.2", 
+    ], 
+    "dhcpv6_servers": [ 
+      "21da:d3:0:2f3b::7", 
+      "21da:d3:0:2f3b::6", 
+    ], 
+    "vlanid": "1000" 
+    } 
+  }
+}
+```
+
+# 2 Modules design
+
+![DHCPv6 Diagram](/doc/DHCPv6_Relay/diagram.png)
+
+## 2.1 DHCP Relay for IPv6 build and runtime dependencies
+
+The DHCP Relay for IPv6 feature, same as the IPv4 version, will be based on the open source project 'isc-dhcp'.
+
+## 2.2 DHCP Relay for IPv6 process in dhcp-relay container
+
+A new process will run in parallel to the other process for IPv4 support.
+The new process will listen to DHCP packets for IPv6 and forward them to the relevant interface according to the configuration.
+For example, from the configuration described on the previous section, the following daemon will start:
+```
+admin@sonic:/# /usr/sbin/dhcrelay -6 -d --name-alias-map-file /tmp/port-name-alias-map.txt -l Vlan1000 -u 21da:d3:0:2f3b::7%Ethernet28 -u 21da:d3:0:2f3b::6%Ethernet28
```

Comment:
The new process will read config data from config db and will not rely on cli.

Reply:
Yes sure, this instance is taken from config DB data, by the template generated for supervisord on dhcp_realy docker.

Reply:
Is it mandatory to specify the upstream interface along with the DHCPv6 server address? The server may not be directly connected to the L3 relay - for example, we could have an IPv6 route to the DHCPv6 server. Will that scenario work? What happens if the IPv6 addresses are removed/re-configured to different upstream interface? Will that require restart of DHCP docker?

Reply:
For the user this configuration is transparent, same as the IPv4 implementation, isc-dhcp-relay will try to relay DHCP packets to all configured servers, the kernel will determine if to send the packet or not depending on the route available in the kernel.
If the server is not directly connected to the L3 interface the DHCP_FORWARD message will be transmitted, if another relay is on the other side, it will forward the message again until it will reach the server and vise versa.
if the IPv6 addresses are removed/re-configured on the L3 up link interfaces than yes, a restart of the service is required.